Makefile

```diff
@@ -85,6 +111,121 @@ prebuild-includes:
 $(GH_PAGES)::
 	@echo "doc available at --> $(DOCS_URL)"
 
+
+# virutalenv setup
+# ----------------
+
+quiet_cmd_pyenv  = PYENV     usage: $ $(PY_ENV_ACT)
+      cmd_pyenv  = \
+	if [ -d "./$(PY_ENV)" -a -x "./$(PY_ENV_BIN)/python" ]; then \
+		echo "PYENV     using virtualenv from $(PY_ENV)"; \
+	else \
+		rm -f $(PY_ENV)/requirements.sha256; \
+		$(PYTHON) -m venv $(VTENV_OPTS) $(PY_ENV); \
+	fi
+
+quiet_cmd_pyenvinstall = PYENV     install $2
+      cmd_pyenvinstall = \
+	if cat $(PY_ENV)/requirements.sha256 2>/dev/null | sha256sum --check --status 2>/dev/null; then \
+		echo "PYENV     already installed"; \
+	else \
+		rm -f $(PY_ENV)/requirements.sha256; \
+		$(PY_ENV_ACT); \
+		python -m pip install $(PIP_VERBOSE) -U pip wheel setuptools; \
+		python -m pip install --use-feature=2020-resolver -r requirements.txt -r requirements-dev.txt && \
+		sha256sum requirements*.txt > $(PY_ENV)/requirements.sha256 ; \
+	fi
```

Comment:
pyenvinstall installs requirements.txt and requirements-dev.txt, so pyyaml is not required.